docker.go

```diff
@@ -603,7 +603,7 @@ func (c *DockerContainer) CopyToContainer(ctx context.Context, fileContent []byt
 		return err
 	}
 
-	err = c.provider.client.CopyToContainer(ctx, c.ID, filepath.Dir(containerFilePath), buffer, types.CopyToContainerOptions{})
+	err = c.provider.client.CopyToContainer(ctx, c.ID, "/", buffer, types.CopyToContainerOptions{})
```

Comment:
Because the path will be preserved by tar itself, we only need to copy tar itself to the base path. When untarred, it will be placed in the containerPath (with an additional directory created if needed).

Here is a [test](https://github.com/testcontainers/testcontainers-go/pull/2049/files#diff-037b83426e676aa3786c08b334b0db7bb70235792961cbe4ed3340bc0a60a7a8R1563) demonstrating this behavior.